docs/operators/deduper.md

```diff
@@ -0,0 +1,523 @@
+    unloading of the buckets to and from the memory. This functions as
+    needed and loads the buckets requested by the Deduper.
+
+This was a very basic introduction to the functioning of the
+Deduper. Following sections will go into more detail on each of the
+components.
+
+Use cases - Basic Dedup
+====================================
+
+Dedup Key
+----------------------
+
+A dedup key is a set of one or more fields in the data tuple which
+acts as the key for the tuples.
+This is used by the deduper to compare tuples to arrive at the
+conclusion on whether two tuples are duplicates.
+
+
+
+Consider an example schema and two sample tuples
+
+`{Name, Phone, Email, Date, State, Zip, Country}`
+
+
+
+Tuple 1:
+```
+{
+  Austin U. Saunders,
+  +91-319-340-59385,
+  ausaunders@semperegestasurna.com,
+  2015-11-09 13:38:38,
+  Texas,
+  73301,
+  United States
+}
+```
+
+
+Tuple 2:
+```
+{
+  Austin U. Saunders,
+  +91-319-340-59385,
+  austin@semperegestasurna.com,
+  2015-11-09 13:39:38,
+  Texas,
+  73301,
+  United States
+}
+```
+
+
+Let us assume that the Dedup Key
+is `{Name, Phone}`. In
+this case, the two tuples are duplicates because the key fields are same
+in both the tuples. However, if we plan to make the Dedup Key as {Phone,
+Email}, then in this case, the two are unique tuples as the
+keys of both tuples do not match.
+
+
+Use case Details
+-----------------------------
+
+Consider the case of de-duplicating a master data set
+which is stored in a file. Further also consider the
+following schema for tuples in the data set.
+
+`{Name, Phone, Email, Date, City, Zip, Country}`
+
+Also consider that we need to identify unique customers from the
+master data set. So, ultimately the output needed for the use case is
+two data sets - Unique Records and Duplicate Records.
+
+As part of configuring the operator for this use case, we need to
+set the following parameters:
+
+-   ***keyExpression*** - This can be set as
+    the primary key which can be used to uniquely identify a Customer.
+    For example, we can set it to “{Name,
+    Email}”
+
+
+The above configuration is sufficient to resolve the use case.
+
+
+Use case - Dedup with Expiry
+=========================================
+
+Motivation
+-----------------------
+
+The Basic Dedup use case is the most straightforward and is
+usually applied when the amount of data to be processed is not huge.
+However, if the incoming data is huge, or even never-ending, it is
+usually not necessary to keep storing all the data. This is because in
+most real world use cases, the duplicates occur only a short distance
+apart. Hence, after a while, it is usually okay to forget the part of
+the history and consider only limited history for identifying
+duplicates, in the interest of efficiency. In other words, we expire
+(ignore) some tuples which are (or were supposed to be) delivered long
+back. Doing so, reduces the load on the Bucket Store which effectively
```

Comment:
"Bucket Store" is not explained in this document; suggest replacing it with generic text like "storage system" or add a few sentences explaining what it is.